h2o-core/src/main/java/water/util/FileUtils.java

```diff
@@ -97,33 +101,74 @@ else if (file.isDirectory()) {
    *  @param fname filename
    *  @return      Found file or null */
   public static File locateFile(String fname) {
-    // When run from eclipse, the working directory is different.
-    // Try pointing at another likely place
+
     File file = new File(fname);
-    if( !file.exists() )
-      file = new File("target/" + fname);
-    if( !file.exists() )
+    if (file.exists()) return file;
+    
+    final Optional<File> fileInPredefinedPath = findFileInPredefinedPath(fname);
+    if (fileInPredefinedPath.isPresent()) return fileInPredefinedPath.get();
+    
+    file = new File("target/" + fname);
+    if (!file.exists())
       file = new File("../" + fname);
-    if( !file.exists() )
+    if (!file.exists())
       file = new File("../../" + fname);
-    if( !file.exists() )
+    if (!file.exists())
       file = new File("../../../" + fname);
-    if( !file.exists() )
+    if (!file.exists())
       file = new File("../target/" + fname);
-    if( !file.exists() )
+    if (!file.exists())
       file = new File(StringUtils.expandPath(fname));
-    if( !file.exists() )
+    if (!file.exists())
       file = null;
     return file;
   }
 
-  private static void check(boolean cond, String msg) throws IOException{
+  /**
+   * @param fileName File name/path to search for in pre-defined search path
+   * @return An {@link Optional} with the file inside, if the H2O_FILES_SEARCH_PATH is defined and the file exists.
+   * Otherwise an empty {@link Optional}. Never null.
+   */
+  private static Optional<File> findFileInPredefinedPath(final String fileName) {
+    Objects.requireNonNull(fileName);
+
+    final String smallDataPathPrefix = System.getenv("H2O_FILES_SEARCH_PATH");
```

Comment:
maybe this can be renamed now to dataPathPrefix ?

Reply:
Yes, this is seriously wrong. Thanks. Copy & paste error.